entry_types/scrolled/package/src/frontend/navigation/AppHeader.js

```diff
@@ -51,11 +58,27 @@ export default function AppHeader() {
     setMobileNavHidden(!mobileNavHidden);
   };
 
-  function handleMenuClick(target) {
-    setActiveChapter(target);
+  function handleMenuClick(chapterLinkId) {
+    setActiveChapterLink(chapterLinkId);
     setMobileNavHidden(true);
   };
 
+  function renderMenuLinks(chapters) {
+    return chapters.map((chapter, index) => {
+      const chapterIndex = index + 1;
+      const chapterLinkId = 'chapterLink' + chapterIndex;
+      return (
+        <li key={index} className={styles.chapterListItem}>
```

Comment:
Also just as aside: Whenever possible it's best to use keys that represent stable identity. React uses keys to determine whether an element has been inserted. If you used the chapter id as key, for example, and then a chapter gets inserted after the first one, React can actually reuse the DOM nodes of all chapters and just inserts one new DOM node. Using an index as a key causes React to to view the inserted chapter as the new way the second chapter now looks. This causes all following chapter nodes to be updated, basically updating the nth chapter with the contents of the (n-1)th.

Reply:
👍